asab/storage/mongodb.py

```diff
@@ -54,19 +54,21 @@ def upsertor(self, collection, obj_id, version=None):
 
 
 	async def list(self, collection:str):
+		documents = []
 		coll = self.Database[collection]
-		ret = await coll.find({})
-		if ret is None:
-			raise KeyError("NOT-FOUND")
-		return ret
+		cursor = coll.find({})
+		async for document in cursor:
+			documents.append(document)
+		return documents
```

Comment:
Hey, there was a reason to return cursor ... don't enumerate here, you will consume a whole memory easily.